tests/postgres_tests/test_indexes.py

```diff
@@ -455,3 +470,140 @@ def test_spgist_parameters(self):
         with connection.schema_editor() as editor:
             editor.remove_index(CharFieldModel, index)
         self.assertNotIn(index_name, self.get_constraints(CharFieldModel._meta.db_table))
+
+    def test_func_index_cast_search_vector(self):
+        index = GinIndex(Cast('field', SearchVectorField()), name='search_vector_cast')
+        with connection.schema_editor() as editor:
+            editor.add_index(TextFieldModel, index)
+            sql = index.create_sql(TextFieldModel, editor)
+        table = TextFieldModel._meta.db_table
+        constraints = self.get_constraints(table)
+        self.assertIn(index.name, constraints)
+        self.assertIn(constraints[index.name]['type'], GinIndex.suffix)
+        self.assertIs(sql.references_column(table, 'field'), True)
+        self.assertIn('::tsvector', str(sql))
+        with connection.schema_editor() as editor:
+            editor.remove_index(TextFieldModel, index)
+        self.assertNotIn(index.name, self.get_constraints(table))
+
+    def test_func_index_search_vector(self):
+        index = GinIndex(SearchVector('scene', 'setting', config='english'), name='search_vector')
+        with connection.schema_editor() as editor:
+            editor.add_index(Scene, index)
+            sql = index.create_sql(Scene, editor)
+        table = Scene._meta.db_table
+        constraints = self.get_constraints(table)
+        self.assertIn(index.name, constraints)
+        self.assertIn(constraints[index.name]['type'], GinIndex.suffix)
+        self.assertIs(sql.references_column(table, 'scene'), True)
+        self.assertIs(sql.references_column(table, 'setting'), True)
+        with connection.schema_editor() as editor:
+            editor.remove_index(TextFieldModel, index)
+        self.assertNotIn(index.name, self.get_constraints(table))
+
+    def test_func_index_with_tablespace(self):
```

Comment:
This is not a PostgreSQL-specific test. I moved it to the `tests.model_indexes`.